torch/_inductor/kernel/flex_attention.py

```diff
@@ -263,29 +282,40 @@ def build_subgraph_buffer(
         # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
         # -- compute scaling constant ---
-        m_ij = tl.maximum(m_i, tl.max(post_mod_scores, 1))
+        row_max = tl.max(post_mod_scores, 1)
+        m_i_new = tl.maximum(m_i, row_max)
 
-        alpha = tl.math.exp2(m_i - m_ij)
-        p = tl.math.exp2(post_mod_scores - m_ij[:, None])
+        alpha = tl.math.exp2(m_i - m_i_new)
+        p = tl.math.exp2(post_mod_scores - m_i_new[:, None])
         if not ROWS_GUARANTEED_SAFE:
-            masked_out_rows = (m_ij == float("-inf"))
+            masked_out_rows = (m_i_new == float("-inf"))
             alpha = tl.where(masked_out_rows, 0, alpha)
             p = tl.where(masked_out_rows[:, None], 0, p)
 
-        # NB: l_i update is pulled up here since it's a bit faster
-        # NB: For headdim=256, it's faster to move it back down to after m_i =
-        # m_ij
+        # -- scale and update acc --
+        acc_scale = l_i * 0 + alpha  # workaround some compiler bug
+        acc *= acc_scale[:, None]
+        acc = tl.dot(p.to(MATMUL_PRECISION), v.to(MATMUL_PRECISION), acc)
+
+        # -- update m_i and l_i --
         l_i = l_i * alpha + tl.sum(p, 1)
-        # # -- scale and update acc --
-        acc = acc * alpha[:, None]
-        v = tl.load(V_block_ptr)
-        acc = tl.dot(p.to(MATMUL_PRECISION), v, acc)
+        m_i = m_i_new
 
-        # -- update m_i
-        m_i = m_ij
         # update pointers
-        K_block_ptr = tl.advance(K_block_ptr, (0, BLOCK_N))
-        V_block_ptr = tl.advance(V_block_ptr, (BLOCK_N, 0))
+        # if start_n < hi - 1:
+        indices_idx = start_n // BLOCKSPARSE_KV_MULTIPLE
+
+        cur_block = tl.load(kv_indices + indices_idx)
```

Comment:
I know this was the original code but I think this is quite janky haha.

cc: @chenyang78 if you have any suggestions on how to write this code nicer.